src/AstGen.zig

```diff
@@ -1722,6 +1722,55 @@ fn structInitExpr(
         }
     }
 
+    // First: The first found field.
+    // Duplicates: A list of following duplicate field names.
+    const DupeNameRecord = struct { first: Ast.TokenIndex, duplicates: ArrayListUnmanaged(Ast.TokenIndex) };
+
+    var duplicate_names = std.AutoHashMap(u32, DupeNameRecord).init(astgen.arena);
+    try duplicate_names.ensureTotalCapacity(@intCast(struct_init.ast.fields.len));
```

Comment:
The `DupeNameRecord` thing was pointless if you never try to clean up the hashmap! The reason I suggested that was so that we can generally free this data from the arena. Not freeing is fine, but in that case you may as well get rid of `DupeNameRecord` again. The alternative is just to deinit `duplicate_names` somewhere - I'd suggest wrapping all the added code in a block (this also helps with readability - it tells us that this is a self-contained piece of logic) and doing a `defer duplicate_names.deinit()`.